numpy/core/src/umath/ufunc_type_resolution.c

```diff
@@ -1218,6 +1218,73 @@ type_reso_error: {
 }
 
 
+NPY_NO_EXPORT int
+PyUFunc_RemainderTypeResolver(PyUFuncObject *ufunc,
+                                NPY_CASTING casting,
+                                PyArrayObject **operands,
+                                PyObject *type_tup,
+                                PyArray_Descr **out_dtypes)
+{
+    int type_num1, type_num2;
+    int i;
+    const char *ufunc_name = ufunc_get_name_cstr(ufunc);
+
+    type_num1 = PyArray_DESCR(operands[0])->type_num;
+    type_num2 = PyArray_DESCR(operands[1])->type_num;
+
+    /* Use the default when datetime and timedelta are not involved */
+    if (!PyTypeNum_ISDATETIME(type_num1) && !PyTypeNum_ISDATETIME(type_num2)) {
+        return PyUFunc_DefaultTypeResolver(ufunc, casting, operands,
+                    type_tup, out_dtypes);
+    }
+    if (type_num1 == NPY_TIMEDELTA) {
+        if (type_num2 == NPY_TIMEDELTA) {
```

Comment:
Why not just write this:

``
if  (type_num1 == NPY_TIMEDELTA && type_num2 == NPY_TIMEDELTA) {
    // your code
}
else {
    return PyUFunc_DefaultTypeResolver(...)
}
```

Reply:
It is all just copy-paste from other type resolvers. I was planning to leave room for implementing `mq` and `md` remainders, where there would be other switches to handle `type_num2` on a case by case basis, so check `type_num1` but multiple checks on `type_num2`.